OpenRA.Mods.Common/Traits/World/HierarchicalPathFinderOverlay.cs

```diff
@@ -0,0 +1,133 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Commands;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[TraitLocation(SystemActors.World)]
+	[Desc("Renders a debug overlay showing the abstract graph of the hierarchical pathfinder. Attach this to the world actor.")]
+	public class HierarchicalPathFinderOverlayInfo : TraitInfo, Requires<PathFinderInfo>
+	{
+		public readonly string Font = "TinyBold";
+		public readonly Color GroundLayerColor = Color.DarkOrange;
+		public readonly Color CustomLayerColor = Color.Blue;
+		public readonly Color GroundToCustomLayerColor = Color.Purple;
+		public readonly Color AbstractNodeColor = Color.Red;
+
+		public override object Create(ActorInitializer init) { return new HierarchicalPathFinderOverlay(this); }
+	}
+
+	public class HierarchicalPathFinderOverlay : IRenderAnnotations, IWorldLoaded, IChatCommand
+	{
+		const string CommandName = "hpf";
+		const string CommandDesc = "toggles the hierarchical pathfinder overlay.";
+
+		readonly HierarchicalPathFinderOverlayInfo info;
+		readonly SpriteFont font;
+
+		public bool Enabled { get; private set; }
+
+		public Locomotor Locomotor { get; set; }
```

Comment:
Can we please include a comment here explaining that this is assigned by the UI code, or falls back to the user selection if null?